### PR TITLE
fix: paginate boards

### LINKED ISSUE #65: only first 50 boards are fetched
**Desktop (please complete the following information):**
 - OS: macOS ventura 13.5.2 (22G91)
 - Terminal kitty
 - Architecture: arm
 - Fjira version: 0.12.1

**Describe the bug**
if you have more than 50 boards, you only get the first 50

**To Reproduce**
0. have more than 50 boards
1. try to find a board that starts with Z

**Expected behavior**
either get all the boards or do some fancy search for boards when the user starts typing to search for a board

**Additional notes**
https://github.com/mk-5/fjira/blob/b5db5b8b775baa07013d8af45d947d5b749e925d/internal/jira/jira_boards.go#L18 makes a single query
it gets the pagination data https://github.com/mk-5/fjira/blob/b5db5b8b775baa07013d8af45d947d5b749e925d/internal/jira/jira.go#L131 but ignores it
we just need to send the `startAt` query param in a loop https://developer.atlassian.com/cloud/jira/software/rest/intro#introduction#pagination

### PR DESCRIPTION
**Pull request by raylu**
fixes #65

### CODE REVIEW

**Comment by mk-5** [comment]
@raylu pls rebase to the latest master in order to have CI running

**Comment by mk-5** [comment]
@raylu can you rebase that one as well? I've seen that on the second PR we have a valid pipeline now

File: internal/jira/jira_boards.go [hunk 1]
```diff
@@ -12,19 +12,33 @@ const (
...
-	if err != nil {
-		return nil, err
-	}
-	var result BoardsResponse
-	err = json.Unmarshal(resultBytes, &result)
-	if err != nil {
-		return nil, err
+	params := &findBoardsQueryParams{ProjectKeyOrId: projectKeyOrId, StartAt: 0}
+	var boards []BoardItem
+	for {
```
Comment (mk-5) [inline internal/jira/jira_boards.go:21]: in theory ... that loop could run infinite number of time. It should not happen of course - but in theory is possible. What do you think about some limitation? like "max-rounds" "max-depth" "max-iterations"? just to break that for loop .. after N iterations.

Comment (raylu) [reply internal/jira/jira_boards.go:21]: seems pointlessly defensive
what kind of error are you going to throw when the jira API breaks its pagination contract?

Comment (mk-5) [reply internal/jira/jira_boards.go:21]: any kind of error is better then infinitive loop, right? 😄

Comment (mk-5) [reply internal/jira/jira_boards.go:21]: but in general you are right - it seems to be pointlessly defensive. All right, I'm gonna to merge it

**Pull request merged**